crates/rspack_core/src/normal_module_factory.rs

```diff
@@ -666,9 +659,7 @@ impl NormalModuleFactory {
         }
       }
     }
-    resolved
-      .map(|r| r.merge_by_dependency(dependency_type))
-      .map(Arc::new)
+    resolved.map(Arc::new)
```

Comment:
the merge is done at here https://github.com/web-infra-dev/rspack/blob/ff40041f2d2a864db21c1c1bfbe1d4c924007f81/crates/rspack_core/src/resolver/resolver_impl.rs#L201, and for the `worker-alias` test case it's actually have nothing related with `byDependency`, there isn't something like this in the config:

```js
module: {
  rules: [{
    test: /\.js/,
    resolve: {
      byDependency: {
        worker: ...
      }
    }
  }]
}
```